DataFormats/Detectors/TPC/include/DataFormatsTPC/CompressedClusters.h

```diff
@@ -0,0 +1,61 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file CompressedClusters.h
+/// \brief Container to store compressed TPC cluster data
+/// \author David Rohr
+
+#ifndef ALICEO2_DATAFORMATSTPC_COMPRESSED_CLUSTERS_H
+#define ALICEO2_DATAFORMATSTPC_COMPRESSED_CLUSTERS_H
+
+#include "GPUCommonRtypes.h"
+
+namespace o2
+{
+namespace TPC
+{
+struct CompressedClusters {
+  unsigned int nTracks = 0;
+  unsigned int nAttachedClusters = 0;
+  unsigned int nUnattachedClusters = 0;
+
+  unsigned short* qTotA = nullptr;        // [nAttachedClusters]
+  unsigned short* qMaxA = nullptr;        // [nAttachedClusters]
+  unsigned char* flagsA = nullptr;        // [nAttachedClusters]
+  unsigned char* rowDiffA = nullptr;      // [nAttachedClusters - nTracks]
+  unsigned char* sliceLegDiffA = nullptr; // [nAttachedClusters - nTracks]
+  unsigned short* padResA = nullptr;      // [nAttachedClusters - nTracks]
+  unsigned int* timeResA = nullptr;       // [nAttachedClusters - nTracks]
+  unsigned char* sigmaPadA = nullptr;     // [nAttachedClusters]
+  unsigned char* sigmaTimeA = nullptr;    // [nAttachedClusters]
+
+  char* qPtA = nullptr;            // [nTracks]
+  unsigned char* rowA = nullptr;   // [nTracks]
+  unsigned char* sliceA = nullptr; // [nTracks]
+  unsigned int* timeA = nullptr;   // [nTracks]
+  unsigned short* padA = nullptr;  // [nTracks]
+
+  unsigned short* qTotU = nullptr;     // [nUnattachedClusters]
+  unsigned short* qMaxU = nullptr;     // [nUnattachedClusters]
+  unsigned char* flagsU = nullptr;     // [nUnattachedClusters]
+  unsigned short* padDiffU = nullptr;  // [nUnattachedClusters]
+  unsigned int* timeDiffU = nullptr;   // [nUnattachedClusters]
+  unsigned char* sigmaPadU = nullptr;  // [nUnattachedClusters]
+  unsigned char* sigmaTimeU = nullptr; // [nUnattachedClusters]
+
+  unsigned short* nTrackClusters = nullptr;  // [nTracks]
+  unsigned int* nSliceRowClusters = nullptr; // [36 * 152]
+
+  ClassDefNV(CompressedClusters, 1);
```

Comment:
This will generate a wrong streamer since the persistent arrays dimensions are not given according to rootcling rules e.g. ``char* qPtA = nullptr; //[nTracks]``. If this class never fill be root-serialized, better use ``ClassDefNV(CompressedClusters, 0);``, otherwise remove space in ``// [`` and use only its data members of constants (no expressions) for dimensions.

Reply:
@shahor02 : Thanks, I was actually not sure if this would work and wanted to check it at some time. I'll adapt it.
Do you by chance know if it works with constexpr static data members as constants? I didn't find anything about it in the ROOT documentation.

Reply:
@davidrohr : no, one should either use a persistent data member to indicate the expected size:
````
unsigned int mNSlots = 36 * 152;
unsigned int *nSliceRowClusters = nullptr; //[mNSlots]
````
or fixed size array e.g.
````
static constexp int  NSlots = 36 * 152;
unsigned int nSliceRowClusters[NSlots];
````